waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -814,15 +814,17 @@
 
 (defn attach-envoy-sidecar
   "Attaches envoy sidecar to replicaset"
-  [replicaset reverse-proxy service-description base-env service-port port0]
+  [replicaset reverse-proxy {:strs [health-check-port-index] :as service-description} base-env service-port port0]
   (update-in replicaset
     [:spec :template :spec :containers]
     conj
     (let [{:keys [cmd resources image]} reverse-proxy
           user-env (:env service-description)
           env-map (-> user-env
-                  (merge base-env)
-                  (assoc "PORT0" (str port0) "SERVICE_PORT" (str service-port)))
+                      (merge base-env)
+                      (assoc "HEALTH_CHECK_PORT_INDEX" (str health-check-port-index)
```

Comment:
Are we planning on support health-check-proto too?

Reply:
The existing proof-of-concept doesn't support backend-proto (it just assumes plain-text). Adding that was outside the scope of this particular patch, so I didn't add the health-check-backend-proto either.